src/util/system.cpp

```diff
@@ -711,6 +710,7 @@ const fs::path &GetDataDir(bool fNetSpecific)
             path = "";
             return path;
         }
+        path = fs::canonical(path);
```

Comment:
This isn't in boost 1.47.0, which is our minimum required in older versions. Thus, it is ineligible for use in a bugfix.

Reply:
Hmm, 
```
$ git grep -e 'fs::canonical' | wc -l
5
```

Reply:
lol, looks like #14146 was the culprit, released in v0.18.0.

Might as well just call it a bug in the dependency docs at this point I guess.

utACK

Reply:
According to doc/dependencies.md, our minimum support Boost version is 1.58 ?

Reply:
@sipa that just changed in master (only).

Reply:
This PR is also fixing a bug that's only in master AFAIK?

Reply:
FWIW fs::canonical was introduced in Boost 1.48.

Reply:
> This PR is also fixing a bug that's only in master AFAIK?

Just tested `0.19.1`. It has that bug too.

Reply:
Ah, my mistake.